feat: add optional TLS communication for the metrics exporter on the instances (#4927)
cloudnative-pg/cloudnative-pg: docs/src/monitoring.md

    and will be removed in the future. Please use the label `cnpg.io/cluster`
    instead to select the instances.

### Enabling TLS on the Metrics Port

To enable TLS communication on the metrics port, configure the `spec.monitoring.tls.enable`
setting to `true`. This setup ensures that the metrics exporter uses the same
server certificates as PostgreSQL to secure communication on port 5432.

!!! important
    Changing the `spec.monitoring.tls.enable` value will cause all instances to restart.

>     Plan accordingly to minimize downtime.

Author: "All instances to restart" seems very chaotic, where this is our well known operation of a rolling restart.
And "minimize downtime" ... with a well performed rolling restart, there would be none, right?

Reviewer: What about?

```suggestion
    Changing the `spec.monitoring.tls.enable` value will cause a rollout of all instances in the cluster.
```

>     Changing the `spec.monitoring.tls.enable` value will cause all instances to restart.
>     Plan accordingly to minimize downtime.

Author: ```suggestion
    Changing the `spec.monitoring.tls.enable` value will initiate a rolling restart of the cluster.
```
